plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/errors/VerificationError.kt

```diff
@@ -40,8 +49,17 @@ object ErrorAdapter {
  * as offending node result the call-site of the called method.
  * But the actual info we are interested in is on the pre-condition, contained in the reason's offending node.
  */
-inline fun <reified I> VerificationError.getInfoOrNull(): I? =
+fun <I> VerificationError.getInfoOrNull(): I? =
     Info.fromSilver(result.offendingNode().info).unwrapOr<I> {
         Info.fromSilver(result.reason().offendingNode().info).unwrapOr<I> { null }
     }
 
+/**
+ * If the reason's offending node is a function application, then fetch the info metadata
+ * from the index-th argument.
+ * Otherwise, return no info.
+ */
+fun ErrorReason.extractInfoFromFunctionArgument(argIndex: Int): Info = when (val node = reason.offendingNode()) {
```

Comment:
When is it not an error to use this when the reason is not a func app?

Reply:
Yes, when the reason's offending node is _not_ a `FuncApp` then it would be an error. I think invoking `error` would be better instead of returning `NoInfo`. WDYT?

Reply:
I agree.